src/AutosizeInput.js

```diff
@@ -57,13 +57,11 @@ class AutosizeInput extends Component {
 		this.copyInputStyles();
 		this.updateInputWidth();
 	}
-	componentWillReceiveProps (nextProps) {
-		const { id } = nextProps;
+	componentDidUpdate (prevProps, prevState) {
+		const { id } = prevProps;
 		if (id !== this.props.id) {
-			this.setState({ inputId: id || generateId() });
+			this.setState({ inputId: this.props.id || generateId() });
```

Comment:
More correct to use `getDerivedStateFromProps` for update state based on props

Reply:
Performance difference is despicable. Can't we move ahead and approve the PR? Using this deprecated event is much worse than this detail